src/json-schema/partial/map_state.json

```diff
@@ -49,6 +49,219 @@
+                                    "required": [
+                                        "Bucket", 
+                                        "Key"
+                                    ]
+                                }
+                            },
+                            "required": [
+                                "Resource",
+                                "Parameters"
+                            ]
+                        },
+                        "ItemSelector": {
+                            "type": "object",
+                            "description": "A JSON object that overrides each single element of the item array."
+                        },
+                        "ItemBatcher": {
+                            "type": "object",
+                            "description": "A JSON object that specifies how to batch the items for the ItemProcessor.",
+                            "properties": {
+                                "BatchInput": {
+                                    "type": "object",
+                                    "description": "Specifies a fixed JSON input to include in each batch passed to each child workflow execution."
+                                },
+                                "MaxItemsPerBatch": {
+                                    "type": "integer",
+                                    "description": "An integer that limits the maximum number of items of each sub-array.",
+                                    "minimum": 1
+                                },
+                                "MaxItemsPerBatchPath": {
+                                    "$ref": "common.json#/definitions/referencePath",
+                                    "description": "A reference path to an integer that limits the maximum number of items of each sub-array."
+                                },
+                                "MaxInputBytesPerBatch": {
+                                    "type": "integer",
+                                    "description": "An integer that limits the maximum size in bytes of each sub-array.",
+                                    "minimum": 1
+                                },
+                                "MaxInputBytesPerBatchPath": {
+                                    "$ref": "common.json#/definitions/referencePath",
+                                    "description": "A reference path to an integer that limits the maximum size in bytes of each sub-array."
+                                }
+                            }
+                        },
+                        "ResultWriter": {
+                            "type": "object",
+                            "description": "A JSON object that specifies where to write the results instead of to the Map state's result.",
+                            "properties": {
+                                "Resource": {
+                                    "type": "string",
```

Comment:
Repeated enum suggestion.

Reply:
> Do we need the examples property if we're already using an enum? (repeated)

Not sure which you're referring to. Is it this one from below?